src/components/QuestionnaireForm/QuestionnaireForm.jsx

```diff
@@ -698,9 +699,62 @@ export default class QuestionnaireForm extends Component {
     return c;
   }
 
+  addAuthorToResponse(qr, practitionerRef) {
+    function traverseToItemsLeafNode(item, practitionerRef) {
+      if (!item.item) {
+        return addAuthor(item, practitionerRef);
+      }
+      else {
+        item.item.map(item => {
+          traverseToItemsLeafNode(item, practitionerRef);
+        })
+      }
+    }
+    // url is a string
+    function addAuthor(item, practitionerRef) {
+      var url = "http://hl7.org/fhir/StructureDefinition/questionnaireresponse-author"
+      const urlValRef =
+      {
+        "url": url,
+        "valueReference":
+        {
+          "reference": practitionerRef
+        }
+      }
+      if (item.extension) {
+        // if there is already an extension with author-extension url
+        const completelyFound = item.extension.find(element => element.url === url && element.valueReference.reference === practitionerRef)
+        const urlFound = item.extension.find(element => element.url === url && element.valueReference.reference !== practitionerRef)
+        
+        var urlFoundIndex = item.extension.findIndex(element => element.url === url)
```

Comment:
Only need to get urlFoundIndex in the if(!completelyFound) and if(urlFound). This block is getting a little hairy now :-)